src/app/icd/ICDManager.cpp

```diff
@@ -98,11 +104,60 @@ bool ICDManager::SupportsFeature(Feature feature)
     bool success        = (Attributes::FeatureMap::Get(kRootEndpointId, &featureMap) == EMBER_ZCL_STATUS_SUCCESS);
     return success ? ((featureMap & to_underlying(feature)) != 0) : false;
 #else
-
     return ((mFeatureMap & to_underlying(feature)) != 0);
 #endif // !CONFIG_BUILD_FOR_HOST_UNIT_TEST
 }
 
+void ICDManager::SendCheckInMsgs()
+{
+#if !CONFIG_BUILD_FOR_HOST_UNIT_TEST
```

Comment:
Is this behind the define because of the interactione model? 
This is a problem because we will want to unit-test this.

Reply:
This is not unit-testable because we do need the interaction model. MoreOver, this triggers a MDNS resolve which is not UnitTestable. This part will have to be covered by Integration test only.

Reply:
Can you create a follow issue for this? It would be worth while to investigate how the CASE test work.

Reply:
Was an issue created for this?

Reply:
#30658